editor/plugins/gdextension/gdextension_edit_dialog.cpp

```diff
@@ -0,0 +1,244 @@
+		} else if (compat_max_parts[0] < 4 || (compat_max_parts[0] == 4 && compat_max_parts[1] < 3)) {
+			validation_panel->set_message(MSG_ID_COMPAT_MAX_VERSION, TTR("Compat max version must be at least 4.3.0."), EditorValidationPanel::MSG_ERROR);
+		} else if (IS_BEFORE_CURRENT_VERSION(compat_max_parts)) {
+			validation_panel->set_message(MSG_ID_COMPAT_MAX_VERSION, TTR("Compat max version cannot be before the current Godot version."), EditorValidationPanel::MSG_ERROR);
+		} else {
+			validation_panel->set_message(MSG_ID_COMPAT_MAX_VERSION, TTR("Compatibility maximum version is valid."), EditorValidationPanel::MSG_OK);
+		}
+	}
+}
+
+void GDExtensionEditDialog::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY: {
+			connect(SceneStringName(confirmed), callable_mp(this, &GDExtensionEditDialog::_on_confirmed));
+			get_cancel_button()->connect(SceneStringName(pressed), callable_mp(this, &GDExtensionEditDialog::_on_canceled));
+		} break;
+	}
+}
+
+GDExtensionEditDialog::GDExtensionEditDialog() {
+	get_ok_button()->set_disabled(true);
+	get_ok_button()->set_text(TTR("Save"));
+	set_hide_on_ok(true);
+	set_title(TTR("Edit GDExtension"));
+	set_min_size(Size2(400, 300) * EDSCALE);
+
+	VBoxContainer *vbox = memnew(VBoxContainer);
+	vbox->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	vbox->set_v_size_flags(Control::SIZE_EXPAND_FILL);
+	add_child(vbox);
+
+	ScrollContainer *scroll = memnew(ScrollContainer);
+	scroll->set_custom_minimum_size(Size2(400, 150) * EDSCALE);
+	scroll->set_horizontal_scroll_mode(ScrollContainer::SCROLL_MODE_DISABLED);
+	scroll->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	scroll->set_v_size_flags(Control::SIZE_EXPAND_FILL);
+	vbox->add_child(scroll);
+
+	GridContainer *grid = memnew(GridContainer);
+	grid->set_columns(2);
+	grid->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->set_v_size_flags(Control::SIZE_SHRINK_BEGIN);
+	scroll->add_child(grid);
+
+	// GDExtension file path (read-only).
+	Label *gdextension_path_label = memnew(Label);
+	gdextension_path_label->set_text(TTR("Path:"));
+	gdextension_path_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(gdextension_path_label);
+
+	gdextension_path = memnew(Label);
+	gdextension_path->set_text("res://addons/my_extension/my_extension.gdextension");
+	gdextension_path->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	grid->add_child(gdextension_path);
+
+	// Entry symbol.
+	Label *entry_symbol_label = memnew(Label);
+	entry_symbol_label->set_text(TTR("Entry Symbol:"));
+	entry_symbol_label->set_horizontal_alignment(HORIZONTAL_ALIGNMENT_RIGHT);
+	grid->add_child(entry_symbol_label);
+
+	entry_symbol_edit = memnew(LineEdit);
+	entry_symbol_edit->set_placeholder("libname_library_init");
+	entry_symbol_edit->set_tooltip_text(TTR("Required. The symbol to use as the entry point for the extension."));
```

Comment:
You're using this form only to create the initial `.gdextension` file, not to edit them generally, right?
It might be good to clarify here that you're choosing _the name of a function_, that this can be freely chosen (as long as it's a valid function name), and that it should probably be called something like `<libname>_library_init`.

Reply:
This file is used to edit existing GDExtensions. The class is called `GDExtensionEditDialog`.

Reply:
Oh... I don't think that's a good idea, to be honest.

The .gdextension Files Support comments, and the dependencies section, both of which would be silently destroyed if you overwrote the file with the editor. And it would silently not show those entries otherwise.

I think I'd rather have a text editor with ini highlighting, and perhaps basic content checks for expected keys.

Reply:
(I'm probably biased as a programmer that's very used to editing text files. But a file editor that edits only some parts of the file reliably can be very confusing.)

Reply:
If this is considered dangerous then perhaps the edit dialog should be turned into a read-only GDExtension property viewing dialog, since I still want to show users these things in the GUI. I'd rather keep it editable myself but I'd also welcome feedback from @dsnopek and others.